ksqldb-rest-app/src/test/java/io/confluent/ksql/api/integration/ApiIntegrationTest.java

```diff
@@ -659,6 +660,136 @@ private void shouldFailToExecuteQuery(final String sql, final String message) {
+        Buffer buff = writeStream.getBody();
+        PrintResponse printResponse = new PrintResponse(buff.toString());
+        atomicReference.set(printResponse);
+        return printResponse.getRows().size();
+      } catch (Throwable t) {
+        return -1;
+      }
+    }, is(1));
+
+    PrintResponse printResponse = atomicReference.get();
+    assertThat(printResponse.getRows().get(0), containsString(
+        "key: {\"F1\":[\"a\"]}, value: {\"STR\":\"FOO\",\"LONG\":1,\"DEC\":1.11,\"BYTES_\":\"AQ==\",\"ARRAY\":[\"a\"],\"MAP\":{\"k1\":\"v1\"},\"STRUCT\":{\"F1\":2},\"COMPLEX\":{\"DECIMAL\":0.0,\"STRUCT\":{\"F1\":\"v0\",\"F2\":0},\"ARRAY_ARRAY\":[[\"foo\"]],\"ARRAY_STRUCT\":[{\"F1\":\"v0\"}],\"ARRAY_MAP\":[{\"k1\":0}],\"MAP_ARRAY\":{\"k\":[\"v0\"]},\"MAP_MAP\":{\"k\":{\"k\":0}},\"MAP_STRUCT\":{\"k\":{\"F1\":\"v0\"}}},\"TIMESTAMP\":1,\"DATE\":1,\"TIME\":0}, partition: 0"));
+  }
+
+  @Test
+  public void shouldExecutePrintQueryNoLimit() {
+    // Given:
+    String sql = "PRINT " + TEST_TOPIC + ";";
+
+    // Create a write stream to capture the incomplete response
+    ReceiveStream writeStream = new ReceiveStream(vertx);
+
+    // Make the request to stream a print
+    JsonObject printProperties = new JsonObject().put("auto.offset.reset", "earliest");
+    JsonObject printRequestBody = new JsonObject().put("sql", sql)
+        .put("properties", printProperties);
+    VertxCompletableFuture<HttpResponse<Void>> responseFuture = new VertxCompletableFuture<>();
+
+    // When:
+    client.post("/query-stream").timeout(Long.MAX_VALUE).as(BodyCodec.pipe(writeStream))
+        .sendJsonObject(printRequestBody, responseFuture);
+
+    // Then:
+    AtomicReference<PrintResponse> atomicReference = new AtomicReference<>();
+    assertThatEventually(() -> {
+      try {
+        Buffer buff = writeStream.getBody();
+        PrintResponse printResponse = new PrintResponse(buff.toString());
+        atomicReference.set(printResponse);
+        return printResponse.getRows().size();
+      } catch (Throwable t) {
+        return -1;
+      }
+    }, greaterThanOrEqualTo(6));
+
+    PrintResponse printResponse = atomicReference.get();
+    assertThat(writeStream.isEnded(), is(false));
+    assertThat(printResponse.getError(), is(nullValue()));
+    assertThat(printResponse.getRows().get(5), containsString(
+        "key: {\"F1\":[\"d\"]}, value: {\"STR\":\"BUZZ\",\"LONG\":6,\"DEC\":10.1,\"BYTES_\":\"Bg==\",\"ARRAY\":[\"f\",\"g\"],\"MAP\":{},\"STRUCT\":{\"F1\":null},\"COMPLEX\":{\"DECIMAL\":5.0,\"STRUCT\":{\"F1\":\"v5\",\"F2\":5},\"ARRAY_ARRAY\":[[\"foo\"]],\"ARRAY_STRUCT\":[{\"F1\":\"v5\"}],\"ARRAY_MAP\":[{\"k1\":5}],\"MAP_ARRAY\":{\"k\":[\"v5\"]},\"MAP_MAP\":{\"k\":{\"k\":5}},\"MAP_STRUCT\":{\"k\":{\"F1\":\"v5\"}}},\"TIMESTAMP\":12,\"DATE\":12,\"TIME\":12}, partition: 0"));
+  }
+
+  @Test
+  public void shouldExecutePrintQueryFromEarliestOffsetWithLimit() {
+    // Given:
+    String sql = "PRINT " + TEST_TOPIC + " FROM BEGINNING LIMIT 3;";
```

Comment:
Does the test infrastructure behave any differently whether FROM BEGINNING is provided or not? Looks like it might always return the same rows regardless.

Reply:
It doesn't because the default auto.offset.reset set in the request header is also "earliest" when not set. In other words, the value coming from ksqlConfig.getKsqlStreamConfigProps() is set to "earliest". 

If we we also send the header as we do in other tests, i.e. https://github.com/confluentinc/ksql/blob/8d7f7080dc9742bb1e4bc19904228f2e763e646c/ksqldb-rest-app/src/test/java/io/confluent/ksql/api/integration/ApiIntegrationTest.java#L599, then the test behaves diferently.

Reply:
Are you saying that a `PRINT <topic>;` issued to this new endpoint is the same as `PRINT <topic> FROM BEGINNING;`, and that users have to use `auto.offset.reset=latest` in order to simulate a raw `PRINT <topic>` (without `FROM BEGINNING`)? That's really confusing if so. `PRINT <topic>` wasn't really intended to take `auto.offset.reset` into account. We might want to rethink that decision (i.e., don't set `auto.offset.reset=earliest` by default on this endpoint if the statement is a `PRINT <topic>` statement).